pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -858,6 +867,89 @@ public void removeOwnedServiceUnits(NamespaceName nsName, BundlesData bundleData
+                            if (topicOwnershipListener.getFilter().test(topicName)) {
+                                try {
+                                    topicOwnershipListener.onLoad(topicName);
+                                } catch (Throwable t) {
+                                    LOG.error("Call topic {} ownership lister error", topic, t);
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    LOG.error("Get owned topic list for namespace bundle {} error.", bundle, ex);
+                }
+            });
+        }
+    }
+
+    protected void onNamespaceBundleUnload(NamespaceBundle bundle) {
+        for (NamespaceBundleOwnershipListener bundleOwnedListener : bundleOwnershipListeners) {
+            try {
+                if (bundleOwnedListener.getFilter().test(bundle)) {
+                    bundleOwnedListener.unLoad(bundle);
+                }
+            } catch (Throwable t) {
+                LOG.error("Call bundle {} ownership lister error", bundle, t);
+            }
+        }
+        if (!CollectionUtils.isEmpty(topicOwnershipListeners)) {
+            getOwnedTopicListForNamespaceBundle(bundle).whenComplete((topics, ex) -> {
+                if (ex == null) {
+                    for (String topic : topics) {
+                        TopicName topicName = TopicName.get(topic);
+                        for (TopicOwnershipListener topicOwnershipListener : topicOwnershipListeners) {
+                            if (topicOwnershipListener.getFilter().test(topicName)) {
+                                try {
+                                    topicOwnershipListener.unLoad(topicName);
+                                } catch (Throwable t) {
+                                    LOG.error("Call topic {} ownership lister error", topic, t);
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    LOG.error("Get owned topic list for namespace bundle {} error.", bundle, ex);
+                }
+            });
+        }
+    }
+
+    public void addNamespaceBundleOwnershipListener(NamespaceBundleOwnershipListener... listeners) {
+        checkNotNull(listeners);
+        for (NamespaceBundleOwnershipListener listener : listeners) {
+            if (listener != null) {
+                bundleOwnershipListeners.add(listener);
+            }
+        }
+        getOwnedServiceUnits().forEach(this::onNamespaceBundleOwned);
+    }
+
+    public void addTopicOwnershipListener(TopicOwnershipListener... listeners) {
+        checkNotNull(listeners);
+        for (TopicOwnershipListener listener : listeners) {
+            if (listener != null) {
+                topicOwnershipListeners.add(listener);
+            }
+        }
```

Comment:
for newly added listeners, we need to trigger `#onLoad` for already owned topics, no?

Reply:
yes